subprojects/docs/src/docs/userguide/authoring-builds/build_lifecycle.adoc

```diff
@@ -15,37 +15,52 @@
 [[build_lifecycle]]
 = Build Lifecycle
 
-We said earlier that the core of Gradle is a language for dependency based programming. In Gradle terms this means that you can define tasks and dependencies between tasks. Gradle guarantees that these tasks are executed in the order of their dependencies, and that each task is executed only once. These tasks form a http://en.wikipedia.org/wiki/Directed_acyclic_graph[Directed Acyclic Graph]. There are build tools that build up such a dependency graph as they execute their tasks. Gradle builds the complete dependency graph _before_ any task is executed. This lies at the heart of Gradle and makes many things possible which would not be possible otherwise.
+Gradle is an example of *dependency based programming*: you define tasks and dependencies between tasks.
+Gradle guarantees that these tasks execute in the order of their dependencies.
+Your build scripts and plugins configure this dependency graph.
+This page discusses the phases of the lifecycle Gradle passes through as it interprets those scripts.
+Additionally, this page explains how you can react to events of the lifecycle using notifications.
 
-Your build scripts configure this dependency graph. Therefore they are strictly speaking _build configuration scripts_.
+== Task Graphs
+
+Some build tools assemble a task graph lazily, as they execute tasks.
+Gradle builds the complete task graph eagerly, before executing any task.
+
```

Comment:
```suggestion
Some build tools assemble a task graph as they execute tasks.
Gradle builds the task graph before executing any task.
When you use lazy configuration syntax, Gradle skips configuration for tasks that aren't part of the current build.

```
@eskatos can you verify that this reword is technically correct? I figure it's a good idea to bring up lazy configuration early here.

Reply:
I would not use the word `syntax` here as this is more about the used APIs.
Maybe simply:

> When you use task configuration avoidance, Gradle skips configuration for tasks that aren't part of the current build.

as in https://docs.gradle.org/current/userguide/task_configuration_avoidance.html